Library/Homebrew/cmd/pull.rb

```diff
@@ -126,42 +121,61 @@ def pull
         end
       end
 
+      orig_message = message = `git log HEAD^.. --format=%B`
       if issue && !ARGV.include?("--clean")
         ohai "Patch closes issue ##{issue}"
-        message = `git log HEAD^.. --format=%B`
-
-        if ARGV.include? "--bump"
-          odie "Can only bump one changed formula" unless changed_formulae.length == 1
-          formula = changed_formulae.first
-          subject = "#{formula.name} #{formula.version}"
-          ohai "New bump commit subject: #{subject}"
-          system "/bin/echo -n #{subject} | pbcopy"
-          message = "#{subject}\n\n#{message}"
-        end
-
         # If this is a pull request, append a close message.
         unless message.include? "Closes ##{issue}."
           message += "\nCloses ##{issue}."
-          safe_system "git", "commit", "--amend", "--signoff", "--allow-empty", "-q", "-m", message
         end
       end
 
+      if changed_formulae.empty?
+        odie "cannot bump: no changed formulae found after applying patch" if do_bump
+        is_bumpable = false
+      end
+      if is_bumpable && !ARGV.include?("--clean")
+        formula = changed_formulae.first
+        new_versions = {
+            :stable => formula.stable.nil? ? nil : formula.stable.version.to_s,
+            :devel => formula.devel.nil? ? nil : formula.devel.version.to_s,
+        }
+        orig_subject = message.empty? ? "" : message.lines.first.chomp
+        subject = subject_for_bump(formula, old_versions, new_versions)
+        if do_bump
+          odie "No version changes found for #{formula.name}" if subject.nil?
+          unless orig_subject == subject
+            ohai "New bump commit subject: #{subject}"
+            pbcopy subject
+            message = "#{subject}\n\n#{message}"
+          end
+        elsif subject != orig_subject && !subject.nil?
+          opoo "Nonstandard bump subject: #{orig_subject}"
+          opoo "Subject should be: #{subject}"
+        end
+      end
+
+      if message != orig_message && !ARGV.include?("--clean")
+        safe_system "git", "commit", "--amend", "--signoff", "--allow-empty", "-q", "-m", message
+      end
+
       if fetch_bottles
-        bottle_commit_url = if testing_job
-          bottle_branch = "testing-bottle-#{testing_job}"
-          url
-        else
-          bottle_branch = "pull-bottle-#{issue}"
-          if tap.core_formula_repository?
-            "https://github.com/BrewTestBot/homebrew/compare/homebrew:master...pr-#{issue}"
+        bottle_commit_url =
+          if testing_job
+            bottle_branch = "testing-bottle-#{testing_job}"
+            url
```

Comment:
Sorry if I'm being somewhat picky here, but as far as I can tell you haven't changed the code in this and the following two diff chunks, just reformatted it. That's not good for reviewing and for a clean history, so I'd prefer if this was reverted.